src/main/java/org/adridadou/ethereum/EthereumContractInvocationHandler.java

```diff
@@ -114,8 +114,8 @@ private Constructor lookForNonEmptyConstructor(Class<?> returnType, Object[] res
         throw new IllegalArgumentException("no constructor with arguments found! for type " + returnType.getSimpleName());
     }
 
-    <T> void register(T proxy, Class<T> contractInterface, SoliditySource code, String contractName, EthAddress address, EthAccount account) {
-        final Map<String, CompilationResult.ContractMetadata> contractsFound = compile(code.getSource()).contracts;
+    <T> void register(T proxy, Class<T> contractInterface, SoliditySource code, String contractName, EthAddress address, EthAccount account) throws IOException {
+     	final Map<String, CompilationResult.ContractMetadata> contractsFound =  code.isCompiled() ?  CompilationResult.parse(code.getRawJson()).contracts :   compile(code.getSource()).contracts;
```

Comment:
It does.
And I totally agree with the separation of compiled and source code.

I will cancel this merge and come up with a new one.

btw. the patch for the StandaloneBlockchain is applied. (https://github.com/ethereum/ethereumj/pull/659) I will make a different merge request for this.

greetings.